### DISCUSSION TITLE
Upscaling Integration

### DISCUSSION THREAD

**Discussion by Lightning-Dev** [visioncortex/vtracer#57] [Ideas]
This is such a cool project. Thanks for creating it and spending so much time working on it. I wanted to ask what are the requirements for installing it. Do I need to add any special dependencies like installing rust etc?

I also have ESRGAN and tested images with and without upscaling before using Vtracer and the results are awesome. One idea would be to have an option that upscales the image before tracing it to get the best result. Although that is more of a scripting idea rather than an integration. 

Thanks again!

**Comment by tyt2y3** [comment]
Thank you.

There is no dependency, because in Rust everything is statically linked together. Just grap a binary (or a python wheel).

> upscales the image before tracing it

yes, exactly, I had a vague idea on how to do image denoising and upscaling. But haven't got the time / energy / money to work on it (yet).
